File: swift/sum-of-multiples/sumOfMultiples.swift

class SumOfMultiples {
    static func toLimit(_ limit : Int, inMultiples: Array<Int>) -> Int {
        var sum = 0
        for index in 1..<limit {
            for factor in inMultiples {
                if factor != 0 && index % factor == 0 {
                    sum += index
                    break
                }
            }
        }
        return sum
    }
}

File: elixir/sum-of-multiples/sum_of_multiples.exs
defmodule SumOfMultiples do
  def to(limit, factors) do
    1..(limit - 1)
    |> Enum.filter(fn(x) -> 
      Enum.any?(factors, &(rem(x, &1) == 0)) 
    end) 
    |> Enum.sum
  end
end
